### ISSUE TITLE
Rule Change: `use-baseline` should check CSS units

### ISSUE THREAD

**Issue by ryo-manba** [eslint/css#402]
### What rule do you want to change?

use-baseline

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```css
/* svh became baseline in 2022 */
a { height: 100svh; }

/* dvh became baseline in 2022 */
a { height: 100dvh; }

/* cqw became baseline in 2023 */
a { height: 10cqw; }
```

### What does the rule currently do for this code?

No warnings are reported.

### What will the rule do after it's changed?

The rule will warn when a CSS unit is not a baseline feature at the configured availability level, with `@supports` guard and allowUnits option support.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

This feature was implemented in https://github.com/ryo-manba/stylelint-plugin-use-baseline/pull/112.
Related issue: https://github.com/ryo-manba/stylelint-plugin-use-baseline/issues/111.
